### ISSUE TITLE
MSSQL: SQL Error when creating/editing a group

### ISSUE THREAD

**Issue by waader** [joomla/joomla-cms#6073]
It seems like this is a generic problem. Not matter in what area you want to create or edit a group, whether in articles, user notes, banners, contacts etc.,  you always get a sql error like this one:

Error [Microsoft][SQL Server Native Client 11.0][SQL Server]The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries, and common table expressions, unless TOP or FOR XML is also specified.SQL=SELECT DISTINCT a.id AS value, a.title AS text, a.level, a.published FROM ( SELECT DISTINCT id,title,level,published,parent_id,extension,lft,rgt FROM mmj38_categories WHERE (extension = 'com_content' OR parent_id = 0) AND published IN (0,1) ORDER BY lft ASC) AS a LEFT JOIN [mmj38_categories] AS b ON a.lft > b.lft AND a.rgt < b.rgt LEFT JOIN [mmj38_categories] AS p ON p.id = 19 WHERE NOT(a.lft >= p.lft AND a.rgt <= p.rgt)

Consequently you get an error where a group is assigned to an item eg. when creating/editing an article.

**Comment by alikon** [comment]
Sorry for my dumbness can you clarify a little bit more when this issue arise maybe adding some screenshot<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6073">issues.joomla.org/joomla-cms/6073</a>.</sub>

**Comment by waader** [comment]
Of course:

![joomla34_6073](https://cloud.githubusercontent.com/assets/9270310/6226720/86d5ed4e-b698-11e4-9f4d-1d57554c2359.png)

This is what I get when I create a new article. When you analyze the error message I becomes clear that mssql-server - in my case version 2008R2 - is not able to execute a subquery with an order by clause. Consequently the category tree is empty.

You can go to administrator\components\com_categories\models\fields\categoryedit.php line 97 $subQuery->order('lft ASC') and comment that out. Then the article mask loads without an error but the category tree is not sorted.

If haven´t tested this with later versions of mssql server but I will when I have more time.

**Comment by alikon** [comment]
can you test this dirty trick
```sql
SELECT DISTINCT a.id AS value, a.title AS text, a.level, a.published, a.lft
 FROM ( 
   SELECT DISTINCT id,title,level,published,parent_id,extension,lft,rgt 
     FROM mmj38_categories 
    WHERE (extension = 'com_content' OR parent_id = 0) 
      AND published IN (0,1)   
  ) AS a LEFT JOIN [mmj38_categories] AS b
   ON a.lft > b.lft AND a.rgt < b.rgt 
    LEFT JOIN [mmj38_categories] AS p 
   ON p.id = 19 WHERE 
  NOT(a.lft >= p.lft AND a.rgt <= p.rgt)
  ORDER BY a.lft ASC
```

**Comment by waader** [comment]
The query works but the result set comprises only 11 items versus 25 items before.

**Comment by alikon** [comment]
It was a simply test to understand how mssql works I need a little bit more time for a solution

**Comment by alikon** [comment]
@waader can you test this one #6135

**Comment by wilsonge** [comment]
Closing this as it is solved by #6135
